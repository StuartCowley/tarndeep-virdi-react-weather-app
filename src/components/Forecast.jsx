import React from "react";
import { getDay, dateFromTime } from "./DataFunction";
import WeatherIcon from "./WeatherIcon";

export default function Forecast(props) {
  const { data } = props;
  return (
    <div className="card--forecast">
      <div className="card--daydate">
        <p className="card--date">{getDay(data.date)}</p>
        <p className="card--time">{dateFromTime(data.date)}</p>
      </div>
      <div className="card--temp">
        <h2 className="card--temp-max">{data.temperature.max}°C</h2>
        <h4 className="card--temp-min">{data.temperature.min}°C</h4>
      </div>
      <WeatherIcon icon={data.icon} isLocation={false} />
      <p className="card--description">{data.description}</p>
    </div>
  );
}
