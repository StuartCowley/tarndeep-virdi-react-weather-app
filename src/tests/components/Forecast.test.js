import React from "react";
import { render } from "@testing-library/react";
import Forecast from "../../components/Forecast";

describe("Forecast", () => {
  const data = {
    date: 1619427883,
    temperature: {
      max: 10,
      min: 5,
    },
    description: "sunny",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<Forecast data={data} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct data", () => {
    const { getByText } = render(<Forecast data={data} />);
    expect(getByText("Monday")).toBeInTheDocument();
    expect(getByText("19/01")).toBeInTheDocument();
    expect(getByText("10°C")).toBeInTheDocument();
    expect(getByText("5°C")).toBeInTheDocument();
    expect(getByText("sunny")).toBeInTheDocument();
  });
});
