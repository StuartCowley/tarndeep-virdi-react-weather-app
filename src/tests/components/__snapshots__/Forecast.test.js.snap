// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card--forecast"
  >
    <div
      class="card--daydate"
    >
      <p
        class="card--date"
      >
        Monday
      </p>
      <p
        class="card--time"
      >
        19/01
      </p>
    </div>
    <div
      class="card--temp"
    >
      <h2
        class="card--temp-max"
      >
        10°C
      </h2>
      <h4
        class="card--temp-min"
      >
        5°C
      </h4>
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <img
        alt="Weather Icon"
        class="card--weather-icon"
        data-testid="forecast-icon"
        src="https://openweathermap.org/img/w/Error: Code not Recognised.png"
      />
    </div>
    <p
      class="card--description"
    >
      sunny
    </p>
  </div>
</DocumentFragment>
`;
